api/tests/identity-access-management/unit/domain/models/ClaimsManager.test.js

```diff
@@ -0,0 +1,152 @@
+import { ClaimManager } from '../../../../../src/identity-access-management/domain/models/ClaimManager.js';
+import { expect } from '../../../../test-helper.js';
+
+describe('Unit | Identity Access Management | Domain | Model | ClaimManager', function () {
+  describe('#mapClaims', function () {
+    it('maps all claims from user info', async function () {
+      // given
+      const claimMapping = { firstName: ['given_name'], lastName: ['last_name'] };
+      const additionalClaims = ['foo'];
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping, additionalClaims });
+      const result = claimManager.mapClaims({ given_name: 'Bob', last_name: 'Uncle', foo: 'foo' });
+
+      // then
+      expect(result).to.deep.equal({ firstName: 'Bob', lastName: 'Uncle' });
+    });
+
+    it('maps claims with multiple possible values', async function () {
+      // given
+      const claimMapping = { firstName: ['given_name', 'username'] };
+
+      // when
+      const claimManager = new ClaimManager({ claimMapping });
+      const result = claimManager.mapClaims({ username: 'Bob' });
+
+      // then
+      expect(result).to.deep.equal({ firstName: 'Bob' });
+    });
+
+    it('returns an empty object when user info null', async function () {
+      // when
+      const claimManager = new ClaimManager();
+      const result = claimManager.mapClaims(null);
+
+      // then
+      expect(result).to.deep.equal({});
+    });
+  });
+
+  describe('#pickAdditionalClaims', function () {
+    it('picks additional claims from user data', async function () {
+      // given
+      const claimMapping = { firstName: ['given_name'] };
+      const additionalClaims = ['foo'];
```

Comment:
Ce serait plus parlant si ce test avait plusieurs additionalClaims et avec des noms faisant penser qu'ils sont attendus/nécessaires, par exemple (et adapter le test en conséquence) : 
```suggestion
      const additionalClaims = ['aWantedClaim', 'anotherWantedClaim'];
```